include/unifex/any_unique.hpp

```diff
@@ -222,65 +208,85 @@ struct _with_type_erased_tag_invoke<
   : _with_type_erased_tag_invoke<Derived, CPO, true, Ret(Args...)> {
 };
 
-template <
-    typename Derived,
-    typename CPO,
-    bool NoExcept,
-    typename Sig>
-struct _with_forwarding_tag_invoke;
+template <typename CPO, bool NoExcept = false, typename Sig = typename CPO::type_erased_signature_t>
+struct _dispatch_cpo;
 
-template <
-    typename Derived,
-    typename CPO,
-    bool NoExcept = false,
-    typename Sig = typename CPO::type_erased_signature_t>
-using with_forwarding_tag_invoke =
-    typename _with_forwarding_tag_invoke<Derived, CPO, NoExcept, Sig>::type;
+template <typename CPO, typename Ret, typename... Args>
+struct _dispatch_cpo<CPO, false, Ret(Args...) noexcept>
+  : _dispatch_cpo<CPO, true, Ret(Args...)> {};
 
-template <
-    typename Derived,
-    typename CPO,
-    bool NoExcept,
-    typename Ret,
-    typename... Args>
-struct _with_forwarding_tag_invoke<
-    Derived,
-    CPO,
-    NoExcept,
-    Ret(Args...)> {
+template <typename CPO, bool NoExcept, typename Ret, typename... Args>
+struct _dispatch_cpo<CPO, NoExcept, Ret(Args...)> {
   struct type {
-    friend Ret tag_invoke(
-        _overload::base_cpo_t<CPO> cpo,
-        replace_this_t<Args, Derived>... args) noexcept(NoExcept) {
-      auto& wrapper = extract_this<Args...>{}(args...);
-      auto& wrapped = wrapper.value;
-      return std::move(cpo)(replace_this<Args>::get((Args &&) args, wrapped)...);
+    struct _result {
+      UNIFEX_NO_UNIQUE_ADDRESS _overload::base_cpo_t<CPO> cpo;
```

Comment:
Why do you need to strip back to base_cpo_t here?
Can you just invoke the CPO type inside operator() below?